src/components/Layout.tsx

```diff
@@ -43,16 +43,23 @@ interface NavBarProps {
 
 const Layout: React.FC<LayoutProps> = ({ children }) => {
   const { isAuthenticated, isAdmin, loading, member, isVerified } = useAuth();
+
   const navigate = useNavigate();
   useEffect(() => {
-    if (!isAuthenticated) {
-      return;
-    }
+    let redirectTimeout: NodeJS.Timeout;
 
-    if (!isVerified) {
-      navigate('/auth');
+    if (!loading && isAuthenticated && !isVerified) {
```

Comment:
Instead of using a timer can always just do it the way I had it originally. Set is loading and don't change it until the member is actually loaded. You said you didn't like having the null setting. This is fine but i'd rather have it the other way than using timers.

Reply:
[Here is the PR](https://github.com/swecc-uw/swecc-client/pull/13/files)

Reply:
[And here is you saying it won't redirect you :P](https://github.com/swecc-uw/swecc-client/pull/28#discussion_r1720899402)

Reply:
fixed loading without null, was just updating in too early